src/systems/collision.rs

```diff
@@ -52,18 +52,18 @@ impl<'s> System<'s> for CollisionSystem {
                 collidee_component.collider_name = name_a.name.to_string();
 
                 // Ensure entity stays inside its right bound
-                let new_x = possible_new_x
-                    .min(bounding_rect.right);
+                let new_x = if possible_new_x >= bounding_rect.right{
+                    collidee_name = String::from("Boundary");
```

Comment:
Great work finding the root cause! As you already mentioned, that the pincer is colliding with an object, in this case the collision layer, and not the boundary. And the fix works as the behavior on colliding with the boundary or the collision layer should be the same. But, instead of overriding the `collidee_name` and setting it `Boundary`, I think it would be better to change the pincer collision system, to handle collision with the collision layer just like it is handled for boundary on line 234. Just one more case for `Collision` needs to be added.